server/tests/backends/sql_translator/test_sql_redshift_translator_steps.py

```diff
@@ -0,0 +1,151 @@
+                    f"{case_id} not in the list anymore !"
+                )
+            except Exception as es:
+                logger.info(
+                    f"[-]{es}\n[-] {case_id} was probably already dropped or not available !"
+                )
+
+
+# Translation from Pipeline json to SQL query
+@pytest.mark.parametrize('case_id, case_spec_file_path', test_cases)
+def test_sql_translator_pipeline(case_id, case_spec_file_path):
+    global REDSHIFT_TABLES_TESTS, CLEANER_JOB_DONE
+    case_id = (
+        f"{case_id.replace('/', '')}_toucan_test___{str(int(time.time()))}___{str(randint(1, 100))}"
+    )
+    spec = get_spec_from_json_fixture(case_id, case_spec_file_path)
+    try:
+        with REDSHIFT_CONNEXION.cursor() as curs:
+            curs.execute(f'DROP TABLE IF EXISTS {case_id}')
+        engine = sqlalchemy.create_engine(
+            url=URL.create(
+                drivername='redshift+redshift_connector',
+                host=HOST,
+                port=PORT,
+                database=DATABASE,
+                username=USER,
+                password=PASSWORD,
+            )
+        )
+
+        REDSHIFT_TABLES_TESTS.append(case_id)
+        data_to_insert = pd.read_json(json.dumps(spec['input']), orient='table')
+        data_to_insert.columns = [c.lower() for c in data_to_insert.columns]
+        data_to_insert.to_sql(
+            name=case_id, con=engine, index=False, if_exists='replace', chunksize=1
+        )
+        if 'other_inputs' in spec:
+            for input in spec['other_inputs']:
+                pd.read_json(json.dumps(spec['other_inputs'][input]), orient='table').to_sql(
+                    name=input,
+                    con=engine,
+                    index=False,
+                    if_exists='replace',
+                )
+
+        steps = spec['step']['pipeline']
+        lower_step_columns(steps)
+        steps.insert(0, {'name': 'domain', 'domain': case_id})
+        pipeline = PipelineWithVariables(steps=steps)
+
+        query = translate_pipeline(
+            sql_dialect=SQLDialect.REDSHIFT,
+            pipeline=pipeline,
+            tables_columns={case_id: data_to_insert.columns},
+            db_schema=None,
+        )
+
+        result = pd.read_sql(query, engine)
+        _drop_tables([case_id])
+        if 'other_inputs' in spec:
+            _drop_tables(spec['other_inputs'])
+
+        if 'expected_sql' in spec:
+            result_expected: pd.DataFrame = pd.read_json(
+                json.dumps(spec['expected_sql']), orient='table'
+            )
+        else:
+            result_expected: pd.DataFrame = pd.read_json(
+                json.dumps(spec['expected']), orient='table'
+            )
+        result_expected.columns = [c.lower() for c in result_expected.columns]
+        assert_dataframes_equals(result_expected, result)
+    except KeyboardInterrupt as es:
+        logger.info(es)
+        _drop_tables(REDSHIFT_TABLES_TESTS)
```

Comment:
I wonder if this shouldn't be in a `finally` block, rather than specifically catching here. 

Or maybe a module-scoped fixture ? That'd allow us to parallelize tests. Something like this:

```python
@pytest.fixture(scope='module', autouse=True)
def autodrop_tables():
    try:
        yield
    finally:
        _drop_tables(REDSHIFT_TABLES_TEST)
```

Reply:
nice :+1: